### PR TITLE
Better and more consistent hyphenation

### PR DESCRIPTION
**Pull request by athoelke**
Some grammar fixes for hyphenation inconsistencies.

* "pseudo-random" -> "pseudorandom"
* Hyphenate "key-generation", "key-derivation", or "key-agreement" when they are used as a noun modifier (i.e. key-derivation algorithm), but not when they are the noun itself.

### CODE REVIEW

File: doc/crypto/api/keys/management.rst [hunk 1]
```diff
@@ -17,7 +17,7 @@ New keys can be created in the following ways:
 *   `psa_import_key()` creates a key from a data buffer provided by the application.
 *   `psa_generate_key()` and `psa_generate_key_custom()` create a key from randomly generated data.
 *   `psa_key_derivation_output_key()` and `psa_key_derivation_output_key_custom()` create a key from data generated by a pseudorandom derivation process. See :secref:`kdf`.
-*   `psa_key_agreement()` creates a key from the shared secret result of a key agreement process. See :secref:`key-agreement`.
+*   `psa_key_agreement()` creates a key from the shared secret result of a key-agreement process. See :secref:`key-agreement`.
 *   `psa_pake_get_shared_key()` creates a key from the shared secret result of a password-authenticated key exchange. See :secref:`pake`.
```
Comment (gilles-peskine-arm) [inline doc/crypto/api/keys/management.rst:21]: `git grep -Eon 'key (agreement|derivation|generation) \w+' doc` reports one “key derivation interface”.

File: doc/crypto/api/ops/pake.rst [hunk 2]
```diff
@@ -1490,7 +1490,7 @@ The key pair is derived from the initial SPAKE2+ password prior to starting the
 It is recommended to use a key-stretching derivation algorithm, for example PBKDF2.
 This process can take place immediately before the PAKE operation, or derived at some earlier point and stored by the participant.
 Alternatively, the Verifier can be provisioned with the `PSA_KEY_TYPE_SPAKE2P_PUBLIC_KEY()` for the protocol, by the Prover, or some other agent.
-:numref:`fig-spake2p-reg` illustrates some example SPAKE2+ key derivation flows.
+:numref:`fig-spake2p-reg` illustrates some example SPAKE2+ key-derivation flows.
 
 The resulting SPAKE2+ key-pair must be protected at least as well as the password.
```
Comment (gilles-peskine-arm) [inline doc/crypto/api/ops/pake.rst:1495]: Conversely this should be “key pair”. From `git grep -Eon 'key-\w+ \w+' doc` (I haven't combed the “key-foo and/or/with”).

We also seem to be inconsistent about keystore vs key-store.

**Review by gilles-peskine-arm — APPROVED** [review]
Looks broadly ok to me. There are still a few inaccuracies but it's cosmetic.

File: doc/crypto/api/ops/pake.rst (hunk 2 above)
Comment (athoelke) [reply doc/crypto/api/ops/pake.rst:1495]: I will review key-pair/key pair, public-key/public key and key-store/keystore. For the latter I am inclined to go with "keystore".

**Pull request merged**